language/libra-vm/src/libra_vm.rs

```diff
@@ -604,26 +601,24 @@ impl LibraVM {
         let txn_gas_price = txn_data.gas_unit_price().get();
         let txn_max_gas_units = txn_data.max_gas_amount().get();
         let txn_expiration_time = txn_data.expiration_time();
-        record_stats! {time_hist | TXN_PROLOGUE_TIME_TAKEN | {
-                self.move_vm
-                    .execute_function(
-                        &account_config::ACCOUNT_MODULE,
-                        &PROLOGUE_NAME,
-                        self.get_gas_schedule()?,
-                        chain_state,
-                        &txn_data,
-                        vec![gas_currency_ty],
-                        vec![
-                            Value::u64(txn_sequence_number),
-                            Value::vector_u8(txn_public_key),
-                            Value::u64(txn_gas_price),
-                            Value::u64(txn_max_gas_units),
-                            Value::u64(txn_expiration_time),
-                        ],
-                    )
-                    .map_err(|err| convert_prologue_runtime_error(&err, &txn_data.sender))
-                }
-        }
+        let _timer = TXN_PROLOGUE_SECONDS.start_timer();
```

Comment:
I am confused by all of those `_timer`. Are we saying we are still "computing" them but not reporting them?
Or I guess I am not understanding the mechanism used to stop them, is it that they just go out of scope?

Reply:
Yes, this returns a HistogramTimer that automatically reports the time when it goes out of scope (https://docs.rs/prometheus/0.3.2/prometheus/struct.HistogramTimer.html).